compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUInterfaces.td

```diff
@@ -86,4 +86,31 @@ def IREEGPU_AnyMmaAttr : Attr<Or<[
   let convertFromStorage = "$_self";
 }
 
+def IREEGPU_PromotionAttr
+    : AttrInterface<"PromotionAttr"> {
+  let cppNamespace = "::mlir::iree_compiler::IREE::GPU";
+
+  let description = [{
+    Interface used to control promotion of operands.
```

Comment:
I assume this is restricted to operands of root ops only? Would it be worth documenting?

Reply:
No, it can go on any operation. The designation of an operation as the "root" op is irrelevant after we've finished `[LLVMCPU|LLVMGPU|SPIRV]ConfigureExecutables`.